File: src/main/java/site/hirecruit/hr/domain/auth/service/UserAuthService.java
package site.hirecruit.hr.domain.auth.service;

import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
import site.hirecruit.hr.domain.auth.dto.OAuthAttributes;
import site.hirecruit.hr.domain.auth.dto.SessionUser;

/**
 * 인증(로그인)를 진행하는 서비스
 *
 * @author 정시원
 * @see UserAuthServiceImpl
 */
public interface UserAuthService {

    /**
     * 세션을 이용한 로그인을 진행한다.
     *
     * @return 세션에 저장된 데이터
     */
    SessionUser login(OAuthAttributes oAuthAttributes) throws OAuth2AuthenticationException;
}

File: src/main/java/site/hirecruit/hr/domain/auth/service/UserAuthService.kt
package site.hirecruit.hr.domain.auth.service

import site.hirecruit.hr.domain.auth.dto.OAuthAttributes
import site.hirecruit.hr.domain.auth.model.User

/**
 * 유저의 인증을 담당하는 서비스
 *
 * @author 정시원
 * @version 1.0
 */
interface UserAuthService {

    /**
     * 인증을 진행합니다.
     *
     * @return [User] - 인증된 유저의 정보
     * @throws OAuth2AuthenticationException 인증에 실패할 경우 발생합니다.
     */
    fun authentication(oAuthAttributes: OAuthAttributes): User
}